src/libc/pthread/thread.rs

```diff
@@ -173,11 +174,18 @@ fn pthread_mach_thread_np(env: &mut Environment, thread: pthread_t) -> mach_port
     host_object.thread_id.try_into().unwrap()
 }
 
+// POSIX usleep
+fn usleep(_env: &mut Environment, useconds: u32) {
+    let duration = Duration::from_millis(u64::from(useconds) / 1000);
+    std::thread::sleep(duration);
```

Comment:
I'm probably not going to merge this because it's, unfortunately, not suited to the threading system touchHLE uses right now. Instead of real (host OS) threads, all the threads are handled by a single thread in a co-operative fashion. In order to properly “sleep”, I'd want to do it in a way where the other threads can keep running. I guess I'd need to add a method to the `Environment` type for doing that, where it'll run the other threads for a while and then return to the original thread once the duration is over.

By the way, do you know what duration the app you were testing was sleeping for? I'm curious whether it's doing a short or a long sleep.

Reply:
The app that does this is Doom. It seems to be a very short sleep, only 1ms: https://github.com/id-Software/DOOM-iOS/search?q=usleep